### PR TITLE
LMB-294: Forbid mandatees from changing fraction

### PR DESCRIPTION
**Pull request by JonasVanHoof**
## Description 
When a fraction is added to a mandataris the user can only change the fraction to one it had before or to independent. This limitation can only be in the update mandataris and NOT in the correct mandataris.

## Test

1. Go to a mandataris or create a new one without a fraction
2. Select a fraction (all options should be in the list)
3. Update it again. The fraction list should contain the previous option and independent

### CODE REVIEW

**Comment by JonasVanHoof** [comment]
The logic for limiting the options in the fraction selector when "Wijzig huidige situatie"  is added.
image


**Blocked by:** https://github.com/lblod/frontend-lokaal-mandatenbeheer/pull/166/files#r1560504829

File: app/components/mandatarissen/fractie-selector.js
Comment (bfidlers) [inline app/components/mandatarissen/fractie-selector.js:52]: I find mandataries really confusing. Maybe better to use mandatarissen or maybe mandatees?

File: app/components/mandatarissen/fractie-selector.js
Comment (bfidlers) [inline app/components/mandatarissen/fractie-selector.js:154]: Wondering why you added this? This replaces practically a oneliner, but I don't feel like this really makes it more readable, so I don't really get why. Or was it broken before?

File: app/components/mandatarissen/fractie-selector.js
Comment (bfidlers) [inline app/components/mandatarissen/fractie-selector.js:34]: Do you need a dedicated function for this? Can't you just use this.args.isUpdatingFractie? Also might be better to use another variable name, since isUpdatingFractie is a bit vague, in the other use case, you also update the fractie ... maybe better something like "isUpdatingState"?

File: app/components/mandatarissen/fractie-selector.js
Comment (bfidlers) [inline app/components/mandatarissen/fractie-selector.js:156]: Personally I wouldn't do this, because it seems pretty redundant, but I can live with it.

**Review by bfidlers — CHANGES_REQUESTED** [review]
Left a few comments. 
I would update the variable names since these are pretty confusing. The other comments are more personal preference, so that's your call.
Functionally everything seems fine.

File: app/components/mandatarissen/fractie-selector.js
Comment (JonasVanHoof) [reply app/components/mandatarissen/fractie-selector.js]: Updated the naming

File: app/components/mandatarissen/fractie-selector.js
Comment (JonasVanHoof) [reply app/components/mandatarissen/fractie-selector.js]: You are right the oneliner is better. I think while implementing I used it on multiple places thats why it moved to a spearte method, good one

**Review by bfidlers — APPROVED** [review]
Thanks, looks good!

File: app/components/mandatarissen/fractie-selector.js
Comment (JonasVanHoof) [reply app/components/mandatarissen/fractie-selector.js]: Updated the args value that's indeed some code thats not needed there. 👌🏻 
Hm I thought about to but when checking the actions and button names I came out on `isUpdatingFractie` becasue the other action is `correcting` the fractie. Just using the `isUpdatingState` name will not be enough because the component is always updating the fractie ;p

Let's see on monday what the naming says :D

**Pull request merged**
